assignments/prototype-refactor.js

```diff
@@ -2,39 +2,74 @@
 // Today your goal is to refactor all of this code to use ES6 Classes.
 // The console.log() statements should still return what is expected of them.
 
-function GameObject(options) {
-  this.createdAt = options.createdAt;
-  this.dimensions = options.dimensions;
+class GameObject {
+constructor(options) { 
+	this.createdAt = options.createdAt;
+	this.dimensions = options.dimensions;
 }
+destroy () {
+    return`Object was removed from the game.`
+    }
+}
+  
+  class CharacterStats extends GameObject{
+	constructor(characterStatsOptions){
+	super(characterStatsOptions)
+	//GameObject.call(this, characterStatsOptions)
+	  this.hp = characterStatsOptions.hp;
+	  this.name = characterStatsOptions.name;
+	}
+  takeDamage () {
+      return `${this.name} took damage.` 
+    }
+    
+	}
+  
+  class Humanoid extends CharacterStats{
+		constructor(humanAttr) {
+      super(humanAttr)
+		//  CharacterStats.call(this, humanAttr);
+		  this.faction = humanAttr.faction;
+		  this.weapons = humanAttr.weapons;
+		  this.language = humanAttr.language;
+       
+}
+		greet  () {
+   console.log(`${this.name} offers a greeting in ${this.language}.`) ;
```

Comment:
It would be better to return this instead of logging it to the console here. If you return it, the console.log on line 123 will function correctly.